tests/system/test_cluster/test_agent_groups/test_group_sync_status.py

```diff
@@ -0,0 +1,185 @@
+    groups_created = host_manager.run_command(testinfra_hosts[0], f'{WAZUH_PATH}/bin/agent_groups')
+    for group in groups:
+        if group in groups_created:
+            response = host_manager.make_api_call(host=testinfra_hosts[0], token=host_manager.get_api_token(testinfra_hosts[0]), method='DELETE',
+                                                      endpoint=f'/groups?groups_list={group}')
+            assert response['status'] == 200, f"Failed to delete {group} group: {response}"
+
+@pytest.fixture
+def group_creation():
+    delete_all_groups()
+    for group in groups:
+        response = host_manager.make_api_call(host=testinfra_hosts[0], token=host_manager.get_api_token(testinfra_hosts[0]), method='POST',
+                                                      endpoint='/groups', request_body={'group_id': group})
+        assert response['status'] == 200, f"Failed to create {group} group: {response}"
+    
+@pytest.fixture
+def agent_group_assignation():
+    agent_ids = host_manager.run_command(testinfra_hosts[0], f'cut -c 1-3 {WAZUH_PATH}/etc/client.keys').split()
+    for group in groups:
+        for agent_id in agent_ids:
+            response = host_manager.make_api_call(host=testinfra_hosts[0], token=host_manager.get_api_token(testinfra_hosts[0]), method='PUT',
+                                                      endpoint=f'/agents/{agent_id}/group/{group}')
+            assert response['status'] == 200, f"Failed to assign agent {agent_id} in {group} group: {response}"
+    
+@pytest.fixture
+def delete_group_folder(test_case):
+    groups_created = host_manager.run_command(testinfra_hosts[0], f'{WAZUH_PATH}/bin/agent_groups')
+    if test_case['test_case']['group_deleted'] in groups_created:
+        host_manager.run_command(test_case['test_case']['host'], f"rm -r {WAZUH_PATH}/etc/shared/{test_case['test_case']['group_deleted']} -f")
+    
+@pytest.fixture
+def wdb_query_creator():
+    wdb = read_file(wdb_query)
+    host_manager.modify_file_content(host=testinfra_hosts[0], path=f'{WAZUH_PATH}/wdb-query.py',content=wdb)
+
+def query_database(): 
+    query = 'global sql select group_sync_status from agent;'
+    response= host_manager.run_command(testinfra_hosts[0], f'python3 {WAZUH_PATH}/wdb-query.py "{query}"')
+    return response 
+
+@pytest.fixture
+def kill_initial_syncreq():
+    result = query_database()
+    while 'syncreq' in result:
+        time.sleep(1)
+        result = query_database()
+
+@pytest.fixture
+def first_check():
+    global first_time_check
+    first_time_check = "synced"
+    s_time = 15
+    for i in range(s_time):
+        time.sleep(0.25)
+        result = query_database()
+        if 'syncreq' in result:
+            first_time_check = "syncreq"
+
+@pytest.fixture          
+def second_check():
+    time.sleep(10)
+    global second_time_check
+    second_time_check = "synced"
+    result = query_database()
+    if 'syncreq' in result:
+        second_time_check = "syncreq"    
```

Comment:
This should be part of the test

Reply:
Checked and corrected in https://github.com/wazuh/wazuh-qa/pull/4143